src/connector/src/source/mod.rs

```diff
@@ -20,12 +20,29 @@ pub mod filesystem;
 pub mod google_pubsub;
 pub mod kafka;
 pub mod kinesis;
+pub mod monitor;
 pub mod nexmark;
 pub mod pulsar;
 pub use base::*;
 pub use google_pubsub::GOOGLE_PUBSUB_CONNECTOR;
 pub use kafka::KAFKA_CONNECTOR;
 pub use kinesis::KINESIS_CONNECTOR;
+use risingwave_common::catalog::TableId;
 
 pub use crate::source::nexmark::NEXMARK_CONNECTOR;
 pub use crate::source::pulsar::PULSAR_CONNECTOR;
+
+#[derive(Clone, Copy, Debug, Default)]
+pub struct SourceContext {
```

Comment:
Is `SourceInfo` a better name?

If you think it's a type of `Context`, you can also store `SourceMetrics` here and store the whole Context in an `Arc`.

Reply:
Renamed it to `SourceInfo`